core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -905,6 +909,50 @@ public void deleteTenant(int tenantId, boolean removeFromPersistentStorage)
         }
     }
 
+    /**
+     * Delete all tenant information related to tenant stored in UM tables.
+     *
+     * @param tenantId Id of the tenant
+     * @throws UserStoreException
+     */
+    @Override
+    public void deleteTenantUMData(int tenantId) throws UserStoreException {
+
+        ServerConfigurationService serverConfigurationService = UserStoreMgtDSComponent.getServerConfigurationService();
+
+        try (Connection conn = getDBConnection()) {
+            conn.setAutoCommit(false);
+
+            executeDeleteQuery(conn, DBConstants.DELETE_USER_PERMISSIONS_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, DBConstants.DELETE_ROLE_PERMISSIONS_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, DBConstants.DELETE_PERMISSION_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, JDBCRealmConstants.DELETE_USER_PROPERTY_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, JDBCRealmConstants.DELETE_DOMAIN_FROM_USER_ROLE_BY_TENANT_ID, tenantId);
+            executeDeleteQuery(conn, JDBCRealmConstants.REMOVE_USER_ROLES_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, HybridJDBCConstants.DELETE_ROLES_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, HybridJDBCConstants.DELETE_REMEMBERME_VALUES_BY_TENANT_ID_SQL, tenantId);
+
+            if (Boolean.parseBoolean(serverConfigurationService
+                    .getFirstProperty("Tenant.DeletePrimaryUsersOnTenantDeletion"))) {
+                executeDeleteQuery(conn, JDBCRealmConstants.DELETE_ROLES_BY_TENANT_ID_SQL, tenantId);
+                executeDeleteQuery(conn, JDBCRealmConstants.DELETE_USERS_BY_TENANT_ID_SQL, tenantId);
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Tenant.DeletePrimaryUsersOnTenantDeletion flag is not enabled in carbon.xml. " +
+                            "Users and Roles will not be deleted.");
+                }
+            }
+
+            conn.commit();
+        } catch (SQLException e) {
+            String errorMsg = "An error occurred while deleting um data for tenant: " + tenantId;
+            log.error(e);
```

Comment:
Current practice is either handle exception and log it with appropriate information. OR throw it.  Log and throw is discouraged as it fills up the logs unnecessarily